### PR TITLE
[WIP] jenkins: multiple pipelines

### PR DESCRIPTION
**Pull request by lamping7**
I started off by pulling out the ROS tests. We need to figure out what else should get split out of the original Jenkinsfile. I also refactored the pipeline for the tests in a pretty major way that should make adding or changing the tests pretty easy.

Current issues: Both pipelines still submit status to the same GitHub `continuous-integration/jenkins/branch` check even though the new one is manually setting a different status. This will be a race and the last one will win. By default, you cannot turn this off in Jenkins, allowing us to handle all of it manually. It looks like there is plugin for this that needs to be looked into. https://issues.jenkins-ci.org/browse/JENKINS-35453

### CODE REVIEW

**Comment by dagar** [comment]
This is awesome! Any concerns we should evaluate before pulling the trigger?

File: .ci/Jenkinsfile-SITL_tests [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+				}
+			}
+		} // script
+	} // steps
+} // Build
+
+stage('ROS Tests') {
+	steps {
+		script {
+			def missions = [
```
Comment (dagar) [inline .ci/Jenkinsfile-SITL_tests]: Nice, this is nearly at the point we could define all the missions in json.

**Comment by lamping7** [comment]
From our discussion on slack, I think we have a handle on what we want to do about the check notifications. I'll get https://plugins.jenkins.io/pipeline-githubnotify-step integrated in and we'll go from there.

**Comment by lamping7** [comment]
Using this plugin: https://github.com/steven-foster/github-scm-trait-notification-context allows us to simply set the text through the setup GUI and remove the worry about conflicting notifications.

**Comment by lamping7** [comment]
I should call this a WIP... the next step is to remove the old Jenkinsfile.

**Comment by dagar** [comment]
Great! My only minor concern is transition for other branches (especially old ones like stable) given the location of the Jenkinsfile is configured Jenkins side. Maybe we should leave a top level simple Jenkinsfile for some period? It could contain the style check, clang-tidy, etc.

**Comment by lamping7** [comment]
As long as we don't remove the old "Firmware" job from Jenkins, it doesn't matter. All three of these checks use a new Jenkinsfile associated with new jobs in Jenkins.

So, with this PR and subsequent CI checks after this is merged, the old job isn't used.

Old branches, with the original Jenkinsfile location will continue run as they did and there will be no effect from this PR.

**Comment by lamping7** [comment]
This looks good. I'll give it another review this weekend and look over formatting.

**Comment by lamping7** [comment]
I restructured the build node creation similarly to how the ROS test nodes are generated. This structure should be more clear than the previous method of how a build was defined, making it easier for users to see how to add more. I also combined the different functions that create the build nodes into one.

**Comment by dagar** [comment]
Splitting compile jobs to separate pipeline - https://github.com/PX4/Firmware/pull/10149

**Comment by lamping7** [comment]
Have all the pieces been broken out into separate PRs? Is this good to close?

**Comment by dagar** [comment]
Not exactly the same, but close enough.
Thanks!

**Pull request closed without merging**